invalid memory address if zones.yaml can't be found

If the zones file can not be found, it reports an error but continues and runs into an 

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8f570d]

```
goroutine 1 [running]:
github.com/hm-edu/harica/dns.(*DNSProvider).matchingProvider(0xf020e0?, {0xc000273955, 0x7})
        /home/runner/work/harica/harica/dns/provider.go:95 +0x2d
github.com/hm-edu/harica/dns.(*DNSProvider).Configured(...)
        /home/runner/work/harica/harica/dns/provider.go:70
github.com/hm-edu/harica/cmd.init.func3(0xc00011a600?, {0xa2dde8?, 0x4?, 0xa2ddec?})
        /home/runner/work/harica/harica/cmd/validation.go:92 +0x89d
github.com/spf13/cobra.(*Command).execute(0xeda6a0, {0xc0000bd0e0, 0x12, 0x12})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:989 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0xeda3c0)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/hm-edu/harica/cmd.Execute()
        /home/runner/work/harica/harica/cmd/root.go:19 +0x1a
main.main()
        /home/runner/work/harica/harica/main.go:6 +0xf

```

--- a/cmd/validation.go
+++ b/cmd/validation.go
@@ -42,6 +42,7 @@ var validationCmd = &cobra.Command{
 		dnsProvider, err := axfr.NewDNSProvider(config.dnsConfig)
 		if err != nil {
 			slog.Error("Failed to create dns provider", slog.Any("error", err))
+			return
 		}
 
 		if len(config.domains) > 10 {

